src/services/codefixes/inferFromUsage.ts

```diff
@@ -340,6 +376,27 @@ namespace ts.codefix {
         }
     }
 
+    function inferTypeForThisFromUsage(containingFunction: SignatureDeclaration, program: Program, cancellationToken: CancellationToken) {
+        let searchToken;
+
+        switch (containingFunction.kind) {
+            case SyntaxKind.FunctionExpression:

```

Comment:
can you add a test for FunctionExpression?

Reply:
Ha, turns out `getAssignedName` in `src/compiler/utilities.ts`' `getNameOfDeclaration` didn't account for having a `VariableDeclaration` as a parent. 

`getNameOfDeclaration` is a pretty heavily used public function. From the baseline changes, it seems like this is a slightly helpful change for some error messages & generated code?

Reply:
Yes, although it's interesting that the biggest improvement was in emitted code. I think that JS is not covered that well in our tests.